Assets/Scripts/Combat/Combantants/Combatant.cs

```diff
@@ -32,39 +47,94 @@ public void Start()
     {
         CombatSystem = GameObject.FindGameObjectWithTag("CombatSystem").GetComponent<CombatSystem>();
         Shield.SetActive(false);
+        IsShieldActive = false;
         IsInCombat = false;
 
         HealthBar = Instantiate(HealthBarPrefab);
         HealthBar.transform.SetParent(HealthBarUIPanel.transform);
+        RedBar = HealthBar.GetComponentInChildren<HealthBar>();
+        HealthText = HealthBar.GetComponentInChildren<Text>();
+        RedBar.MaxHealth = MaxHealthPoints;
         HealthBar.SetActive(false);
-
+        
         CurrentHealthPoints = MaxHealthPoints;
     }
 
     public void Update()
     {
         if(IsInCombat)
+        {
             UpdateHealthBar();
+            UpdateShieldBar();
+        }
     }
 
     private void UpdateHealthBar()
     {
-        string healthText;
+        RedBar.NewHealth = CurrentHealthPoints;
+        HealthText.text = $"{CurrentHealthPoints} / {MaxHealthPoints}";
+        
+        UpdateHealthBarPosition();
+    }
 
-        if (CurrentShieldPoints > 0)
-            healthText = IsAlive ?
-                        $"{CurrentHealthPoints} / {MaxHealthPoints} Shield: {CurrentShieldPoints} / {MaxShieldPoints}" :
-                        "I've fallen and can't get up";
+    private void UpdateShieldBar()
+    {
+        if (IsShieldActive && CurrentShieldPoints > 0)
+        {
+            ShieldBar.SetActive(true);
+            BlueBar.NewHealth = CurrentShieldPoints;
+            ShieldText.text = $"{CurrentShieldPoints} / {MaxShieldPoints}";
+        
+            UpdateShieldBarPosition();
+        }
+        else if (CurrentShieldPoints <= 0 && IsShieldActive)
+        {
+            ShieldBar.SetActive(false);
```

Comment:
Should you also IsShieldActive = false here too?

Reply:
No, isShieldActive is used to check if a shield was spawned, I guess I'll change the name